lib/linter/apply-disable-directives.js

```diff
@@ -18,6 +18,76 @@ function compareLocations(itemA, itemB) {
     return itemA.line - itemB.line || itemA.column - itemB.column;
 }
 
+/**
+ * Returns whether a comment is alone on its line.
+ * @param {Comment} comment The comment node which will be deleted.
+ * @param {string} line Contents of the comment's line.
+ * @returns {boolean} Whether the comment is alone on its line.
+ */
+function commentIsAloneOnLine(comment, line) {
+    return comment.type === "Block" ? line.length === comment.value.length + "/**/".length : line.startsWith("//");
+}
```

Comment:
This logic could be wrong in some cases:

```js
/* eslint semi:2 */

const code = `const x = 5
// comment in the code`; // eslint-disable-line semi

```

autofixed to:

```js
/* eslint semi:2 */

const code = `const x = 5

```

The logic for block comments might be also wrong if the comment is multiline and the line where it starts happen to have the same length as the comment.